### PR TITLE
doc: Update the release process

### PR DESCRIPTION
**Pull request by stevenhorsman**
Add a step to wait for the payload publish to complete before running the release action.

### CODE REVIEW

**Review by gkurz — APPROVED** [review]
Problem is that successful build isn't required for all assets in CI and people end up merging PRs that break some asset without noticing until we release. I'm fine with adding an extra step as an effort to mitigate the consequences on the release process but we should still straighten the status of assets.

Thanks @stevenhorsman !

File: docs/Release-Process.md [hunk 1]
```diff
@@ -45,6 +45,14 @@ branch to read only whilst the release action runs.
 > [!NOTE]
 > Admin permission is needed to complete this task.
 
+### Wait for the `VERSION` bump PR payload publish to complete
```
Comment (sprt) [inline docs/Release-Process.md:48]: So if the version bump PR fails CI, rerunning it is the way to go?

Comment (stevenhorsman) [reply docs/Release-Process.md:48]: Yeah, it's either re-run it, or if a fix is required then doing it before the need to mess about with draft releases is easier.

**Review by zvonkok — APPROVED** [review]
LGTM, lesson learned!

**Comment by stevenhorsman** [comment]
> I still need clarification about Gatekeeper. Shouldn't all build targets be not required, aka not run, if we update the .md?

I've pushed a new commit with an update to the required-test rules to try and address this, just as a point of moving us on a bit. Added `do-not-merge` whilst this is tested.

**Comment by stevenhorsman** [comment]
> > I still need clarification about Gatekeeper. Shouldn't all build targets be not required, aka not run, if we update the .md?
> 
> I've pushed a new commit with an update to the required-test rules to try and address this, just as a point of moving us on a bit. Added `do-not-merge` whilst this is tested.

Damn - in the act of updating the require-tests.yaml this PR is now not a doc only change 😢 - I'll make a temp update to try and get around that for testing.

**Comment by stevenhorsman** [comment]
> Damn - in the act of updating the require-tests.yaml this PR is now not a doc only change 😢 - I'll make a temp update to try and get around that for testing.

With my temp commit the gatekeeper job does what I'd expect: https://github.com/kata-containers/kata-containers/actions/runs/12410277581/job/34646671135
(just the WIP and commit message checks fail as we have the `do-not-merge` label and my temp commit doesn't match the correct format). I will spin this change out as a new PR, but hopefully in future the gatekeep will work better for doc only changes.

**Pull request merged**